src/core/Akka.Streams/Stage/GraphStage.cs

```diff
@@ -18,57 +18,61 @@
 
 namespace Akka.Streams.Stage
 {
-    public interface IGraphStageWithMaterializedValue
+    public interface ILogicAndMaterializedValue<out TMaterialized>
```

Comment:
Does it have to be an interface? Where there any downsides of using struct directly? Also, if we implement it as a struct and return and interface we have cost of additional boxing.